app/server/controllers/UserController.js

```diff
@@ -736,4 +736,116 @@ UserController.getStats = function(callback){
   return callback(null, Stats.getUserStats());
 };
 
+UserController.addUserAcceptedQueue = function(id, callback){
+  User.findOneAndUpdate({
+    _id: id,
+    'status.admitted': false
+  },{
+    $set: {
+      'status.queued': Date.now(),
+    }
+  },
+  function(err, user){
+    if(err || !user){
+      return callback(err);
+    }
+    if(user.status.queued){
+      return callback({
+        message: `User id:${id}, email:${user.email} already to acceptance queue`
+      });
+    } else {
+      return callback(null, {
+        message: `User id:${id}, email:${user.email} added to acceptance queue`
+      });
+    }
+  });
+};
+
+UserController.removeUserAcceptedQueue = function(id, callback){
+  User.findOneAndUpdate({
+    _id: id
+  },{
+    $set: {
+      'status.queued': null
+    }
+  },
+  function(err, user){
+    if(err || !user){
+      return callback(err);
+    }
+    if(!user.status.queued){
+      return callback({
+        message: `User id:${id}, email:${user.email} already not in acceptance queue`
+      });
+    } else {
+      return callback(null, {
+        message: `User id:${id}, email:${user.email} deleted from acceptance queue`
+      });
+    }
+  });
+};
+
+UserController.acceptAllInAcceptedQueue = function(admitterEmail, callback){
+  User.update({
+    'status.queued' : {$ne: null},
+    'status.admitted' : false
+  },{
+    $set: {
+      'status.admitted': true,
+      'status.admittedBy': admitterEmail,
+      'status.queued': null
+    }
+  }, {
+    multi: true
+  },
+  function(err, user){
+    if (err || !user) {
+      return callback(err);
+    }
+    Mailer.sendAcceptanceEmail(user.email);
```

Comment:
Just wanted to check on this. Will this function execute for each user or just the last or first user?